aten/src/ATen/native/vulkan/VulkanCommon.h

```diff
@@ -54,6 +54,21 @@ struct ContextConv2D final {
   static constexpr float kMax = std::numeric_limits<float>::infinity();
 };
 
+namespace detail {
+template <typename To, typename From>
+inline constexpr To safe_downcast(From v) {
+  if (std::is_signed<From>::value && std::is_unsigned<To>::value) {
```

Comment:
Also this is more of an advanced topic (just a tip - you don't need to make the change), but since this condition is completely a compile-time condition, you could either use SFINAE in pre C++-17 or `if constexpr` in C++-17 and onward to isolate it.  That would also prevent against compilation errors in case this if block contained a block of code that would only make sense (i.e. was only compilable) if this condition held true but not in general.  This is not the case here.

Reply:
I have updated PR with enable_if construction :

```
template <typename To, typename From>
inline constexpr To safe_downcast_internal(From v) {
  typedef std::common_type_t<From, To> Type;
  constexpr Type min{static_cast<Type>(std::numeric_limits<To>::lowest())};
  const Type value{static_cast<Type>(v)};
  constexpr Type max{static_cast<Type>(std::numeric_limits<To>::max())};
  TORCH_CHECK(min <= v && v <= max, "Cast failed: out of range");
  return static_cast<To>(v);
}

template <typename To, typename From>
inline constexpr bool IsSignedToUnsigned() {
  return std::is_signed<From>::value && std::is_unsigned<To>::value;
}

template <
    typename To,
    typename From,
    std::enable_if_t<IsSignedToUnsigned<To, From>(), bool> = true>
inline constexpr To safe_downcast(From v) {
  TORCH_CHECK(v >= From{}, "Cast failed: negative signed to unsigned");
  return safe_downcast_internal<To, From>(v);
}

template <
    typename To,
    typename From,
    std::enable_if_t<!IsSignedToUnsigned<To, From>(), bool> = true>
inline constexpr To safe_downcast(From v) {
  return safe_downcast_internal<To, From>(v);
}
```